From 859ef5f9fe95037c2dd4491af7adf0420fb0bd6c Mon Sep 17 00:00:00 2001
From: Cheick Traore <cheick.traore-ext@st.com>
Date: Mon, 2 Sep 2024 15:49:03 +0200
Subject: [PATCH 1104/1695] drivers: stm32_adc: Add support for adc on
 stm32mp23

Add support for STM32 ADCs on stm32mp23.
Add a new compatible because the trigger list is different from stm32mp25
and use the same configuration as on stm32mp25.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f270893dba7154cd3573451d06674374ac2e6307 drivers: stm32_adc: Add support for adc on stm32mp23]
Change-Id: Ic5a1c86eb7d6aa63a5e3d6eb460ae01ac736d52b
Signed-off-by: Cheick Traore <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401509
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 core/drivers/adc/stm32_adc.c      | 1 +
 core/drivers/adc/stm32_adc_core.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/core/drivers/adc/stm32_adc.c b/core/drivers/adc/stm32_adc.c
index dd78b129f..9b651227c 100644
--- a/core/drivers/adc/stm32_adc.c
+++ b/core/drivers/adc/stm32_adc.c
@@ -1381,6 +1381,7 @@ static const struct stm32_adc_cfg stm32mp25_adc_cfg = {
 static const struct dt_device_match stm32_adc_match_table[] = {
 	{ .compatible = "st,stm32mp13-adc", .compat_data = &stm32mp13_adc_cfg },
 	{ .compatible = "st,stm32mp21-adc", .compat_data = &stm32mp21_adc_cfg },
+	{ .compatible = "st,stm32mp23-adc", .compat_data = &stm32mp25_adc_cfg },
 	{ .compatible = "st,stm32mp25-adc", .compat_data = &stm32mp25_adc_cfg },
 	{ }
 };
diff --git a/core/drivers/adc/stm32_adc_core.c b/core/drivers/adc/stm32_adc_core.c
index 0479806ef..bdde0de14 100644
--- a/core/drivers/adc/stm32_adc_core.c
+++ b/core/drivers/adc/stm32_adc_core.c
@@ -447,6 +447,9 @@ static const struct dt_device_match stm32_adc_core_match_table[] = {
 	{ .compatible = "st,stm32mp21-adc-core",
 	  .compat_data =  &stm32mp25_adc_core_cfg
 	},
+	{ .compatible = "st,stm32mp23-adc-core",
+	  .compat_data =  &stm32mp25_adc_core_cfg
+	},
 	{ .compatible = "st,stm32mp25-adc-core",
 	  .compat_data =  &stm32mp25_adc_core_cfg
 	},
-- 
2.47.2

